### PR TITLE
iamy 2.0.0 (new formula)

### PR DESCRIPTION
**Pull request by orien**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

[IAMy](https://github.com/99designs/iamy) is a command line tool for managing AWS IAM users, policies, groups, and roles. Allows you to use an "Infrastructure as Code" model to manage your IAM configuration.

### CODE REVIEW

File: Formula/iamy.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    dir = buildpath/"src/github.com/99designs/iamy"
+    dir.install buildpath.children
+    cd dir do
+      system "glide", "install"
+      system "go", "build", "-o", bin/"iamy", "-ldflags", "-X main.Version=v#{version}"
+    end
+  end
+
+  test do
+    system bin/"iamy", "--help"
```
Comment (ilovezfs) [inline Formula/iamy.rb]: This will need to be a test that verifies the functionality of the software.

**Review by ilovezfs — CHANGES_REQUESTED** [review]

File: Formula/iamy.rb (hunk 1 above)
Comment (orien) [inline Formula/iamy.rb]: I agree. However it is troublesome. The tool has two commands: `pull` and `push`. `pull` connects to an AWS account and writes the current IAM configuration to a YAML representation. Conversely, `push` loads IAM configuration from YAML files and applies it to an AWS account. These operations cannot be exercised without credentials to an active AWS account. Further, I cannot see any automated tests in the source repository.

@ilovezfs What do you suggest in this case?

Comment (ilovezfs) [inline Formula/iamy.rb]: Something like this should be OK:
```
    output = shell_output("#{bin}/iamy pull 2>&1", 1)
    assert_match "Can't determine the AWS account", output
```

Comment (orien) [inline Formula/iamy.rb]: Thanks @ilovezfs. Will do.

Comment (orien) [inline Formula/iamy.rb]: Although this will fail for users that have AWS credentials set in their environment.

Comment (ilovezfs) [inline Formula/iamy.rb]: You can unset them.
```
ENV.delete "AWS_ACCESS_KEY"
ENV.delete "AWS_SECRET_KEY"
```
is probably sufficient.

Comment (orien) [inline Formula/iamy.rb]: Thanks @ilovezfs. Your recommendation has been applied.

**Review by ilovezfs — APPROVED** [review]

**Comment by ilovezfs** [comment]
Thanks for your first contribution to Homebrew, @orien! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request closed without merging**